src/ngraph/node.hpp

```diff
@@ -245,6 +245,13 @@ namespace ngraph
         /// Set device placement
         void set_placement_index(size_t placement);
 
+        std::unordered_set<std::string> get_provenance_tags() const;
+        void add_provenance_tag(const std::string& tag);
+        size_t remove_provenance_tag(const std::string& tag);
```

Comment:
I don't think there's any need to return the `size_t` returned by `erase` --- let's make it `void`.